core/src/main/java/org/elasticsearch/index/query/EmptyQueryBuilder.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.Query;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+/**
+ * A {@link QueryBuilder} that is a stand in replacement for an empty query clause in the DSL.
+ * The current DSL allows parsing inner queries / filters like "{ }", in order to have a
+ * valid non-null representation of these clauses that actually do nothing we can use this class.
+ *
+ * This builder has no corresponding parse because it should not be allowed in the query DSL. It can
+ * be used internally when parsing nested queries to signal parent queries that the this query should be
+ * ignored upstream.
+ */
+public class EmptyQueryBuilder extends AbstractQueryBuilder<EmptyQueryBuilder> {
```

Comment:
I was thinking about NoOp, but the query is not really an operation, an from something called NoOp I'd expect it to be ignored completely in its usual context. This builder returns `null`-Query in `toQuery()` which needs to be handled, thats why I think NoOp is a bit misleading. I initally called it NullQueryBuilder, but that sounds too technical and a bit scary. 
It would be great if there was an empty Lucene-Query that would be ignored by enclosing queries, then we could return that instead of null in `toQuery()`, but I couldn't finde one.